packages/abi-coder/src/coders/b256.ts

```diff
@@ -12,7 +12,7 @@ export class B256Coder extends Coder<string, string> {
   encode(value: string): Uint8Array {
     let encodedValue;
     try {
-      encodedValue = arrayify(value);
+      encodedValue = getBytesCopy(value);
```

Comment:
`getBytesCopy` has been used instead of `getBytes` as it's functionality is closer to that of the previous `arrayify`. `arrayify` always returns a new typed array representing the initially passed value, `getBytes` will mutate the passed value. This posed problems where we passed in a `Buffer` to instantiate program types, as a `Buffer` was then returned by `getBytes` which would throw type errors further along the flow. Hence `getBytesCopy` should be used to ensure a byte array is returned.